arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -448,6 +469,14 @@ public String toString() {
 
     @Override
     public void setBatchSize(int batchSize) {
+      this.batchSize = (batchSize == 0) ? DEFAULT_BATCH_SIZE : batchSize;
+    }
+
+    @Override
+    public void close() {
```

Comment:
I think we did not close the vector before.

Reply:
I don't think that it is the responsibility of the reader to close vectors, since vectors are not owned by readers.

Reply:
I matched the logic of the main reader where we have this:

```
  @Override
  public void close() {
    if (vec != null) {
      vec.close();
    }
  }
```

Looks like we are closing `FieldVector` for regular reads.

Reply:
We should probably remove that implementation of `close`. The vectorized buffers live outside of the readers. Close should be used to close file resources, not data in memory resources.

Reply:
Yeah, looks like `ColumnarBatchReader` does close each vector too. I'll change it.